runtime/reconcilers/model.go

```diff
@@ -1547,6 +1561,75 @@ func (r *ModelReconciler) shouldTrigger(ctx context.Context, self *runtimev1.Res
 	}
 }
 
+// execModelTest runs a single model test and returns an error if it fails. Accepts partitionData for partition-level tests.
+func (r *ModelReconciler) execModelTest(ctx context.Context, self *runtimev1.Resource, test *runtimev1.ModelTest, partitionData map[string]any) error {
+	args := map[string]any{"limit": 1}
+	if partitionData != nil {
+		args["partition"] = partitionData
+		args["split"] = partitionData // Deprecated: use "partition" instead
```

Comment:
Can remove this line. No need to keep backwards compatibility here because tests are a new feature.